packages/next/plugins/with-nx.ts

```diff
@@ -237,6 +234,24 @@ function withNx(
           nextConfig.distDir && nextConfig.distDir !== '.next'
             ? joinPathFragments(outputDir, nextConfig.distDir)
             : joinPathFragments(outputDir, '.next');
+      } else {
+        // if we don't have outputPath set, we are probably running a build using run-commands.
+        // So we need to read distDir from the project's next.config.js. and generate the dist relative to workspace root.
+        // Here we assume the cwd is the projectRoot called from plugins @nx/next/plugin PCV3.
+
+        const nextJSProjectRoot = findNextConfigPath(projectDirectory);
```

Comment:
I wonder if we want to do this, without our executor the dist folder would just contain `dist/<app>/.next` right? The `public`, `next.config.js`, `package.json` are missing?

If we only write `.next` to dist, perhaps it's better to leave it as the default `<project-src>/.next`? That way the source folder is deployable, otherwise the `dist` directory isn't deployable anyway.